### PR TITLE
MODINVOICE-77 Invoice and invoiceLine schema updates

### PR DESCRIPTION
**Pull request by pautau-s**
## Purpose
[MODINVOICE-77](https://issues.folio.org/browse/MODINVOICE-77) Make necessary changes to the schemes for invoice and invoiceLine.

## Approach
- adding `accountingCode` and `accountingNumber` to invoice line schema
- making `source` an enum instead of uuid in invoice.json
- adding required field  `accountingCode` to invoice.json

#### TODOS and Open Questions

- [ ]  Merge dependent PR's for mod-invoice-storage, mod-invoice with this
- [ ]  Merge UI changes for [UINV-45](https://issues.folio.org/browse/UINV-45) along with the PR's above

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Examples exist for all schemas
  - [x] Descriptions exist for all schema properties
  - [x] All schemas pass raml-cop linting
- Does this introduce breaking changes?
  - [x] Were there any schema changes?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [x] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [x] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: mod-invoice-storage/examples/invoice_line_collection.sample [hunk 1]
```diff
@@ -2,6 +2,8 @@
   "invoiceLines": [
     {
       "id": "e0d08448-343b-118a-8c2f-4fb50248d672",
+      "accountingCode" : "0206",
+      "accountNumber":"GOBI-HIST-12",
```
Comment (javalkar-varun) [inline mod-invoice-storage/examples/invoice_line_collection.sample]: Nitpick: Need space between `:` and `"GOBI..."`

File: mod-invoice-storage/examples/invoice_line.sample [hunk 2]
```diff
@@ -1,5 +1,7 @@
 {
   "id": "2be1e551-be27-4b03-b770-30e5e8211d19",
+  "accountingCode": "0206",
+  "accountNumber":"GOBI-HIST-12",
```
Comment (javalkar-varun) [inline mod-invoice-storage/examples/invoice_line.sample]: Nitpick: Need space between `:` and `"GOBI..."`

File: mod-invoice-storage/schemas/invoice.json [hunk 3]
```diff
@@ -94,7 +98,11 @@
     "source": {
       "description": "This does not denote a user ID but describes how the record was created. Eg. Manually, EDIFACT, MARC etc",
       "type": "string",
-      "pattern": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$"
+      "enum": [
```
Comment (KVupp) [inline mod-invoice-storage/schemas/invoice.json]: The description has examples not part of this enum. Please update

File: mod-invoice-storage/schemas/invoice_line.json [hunk 4]
```diff
@@ -8,6 +8,14 @@
       "type": "string",
       "pattern": "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[1-5][0-9a-fA-F]{3}-[89abAB][0-9a-fA-F]{3}-[0-9a-fA-F]{12}$"
     },
+    "accountingCode": {
+      "description": "Number that represents the vendor is an external accounting system which may include details like \"Address code\" in it. This is the number as stated on the invoice",
```
Comment (KVupp) [inline mod-invoice-storage/schemas/invoice_line.json]: > This is the number as stated on the invoice

This description doesn't seem to match the interface fields sheet, which says that the value is pulled from accountNumber which is inturn pulled from either POL/ based on selected vendor. Either way probably doesn't come from invoice

**Comment by KVupp** [comment]
Just an observation, both the UI stories mentioned in the TODO , do not have Acceptance Criteria to make changes to the **source** field, where will that change be handled?

**Comment by pautau-s** [comment]
> Just an observation, both the UI stories mentioned in the TODO , do not have Acceptance Criteria to make changes to the **source** field, where will that change be handled?

Today after the meeting we discussed it with Aliaksei, and tomorrow we're going to create a UI ticket to handle it.

**Pull request merged**